sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -44,7 +44,7 @@ case class PartitionedFile(
     filePath: String,
     start: Long,
     length: Long,
-    locations: Array[String] = Array.empty) {
+    @transient locations: Array[String] = Array.empty) {
```

Comment:
this is not for `FileScanRDD.filePartitions`, this is for `FilePartition`s that sent by scheduler. The location is only useful during planning, we should not send it to executors.